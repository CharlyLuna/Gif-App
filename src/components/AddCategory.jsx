import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onValueChange = ({ target }) => setInputValue(target.value);

  const onSubmit = (event) => {
    const newInputValue = inputValue.trim();
    event.preventDefault();
    if (newInputValue.length <= 1) return;

    // setCategories(inputValue);
    onNewCategory(newInputValue);
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit} aria-label='form'>
      <input
        type='text'
        placeholder='Search gifs'
        value={inputValue}
        onChange={onValueChange}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
