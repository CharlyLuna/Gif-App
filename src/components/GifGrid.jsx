import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h2>{category}</h2>
      {isLoading && <h2>Loading...</h2>}
      <div className='card-grid'>
        {images.map((img) => (
          <GifItem key={img.id} {...img} />
        ))}
      </div>
    </>
  );
};

GifGrid.propTypes = {
  category: PropTypes.string.isRequired,
};
