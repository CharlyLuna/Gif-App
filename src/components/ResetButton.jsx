export const ResetButton = ({ onReset }) => {
  return <button onClick={onReset}>Reset</button>;
};
