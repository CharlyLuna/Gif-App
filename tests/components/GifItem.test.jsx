import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe("GifItem tests", () => {
  const title = "Some title";
  const url = "https://some-random-url.com/img";

  test("should match the snapshot", () => {
    const { container } = render(<GifItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test("should show the image with the url and the ALT indicated", () => {
    render(<GifItem title={title} url={url} />);

    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("should show the title in the component", () => {
    render(<GifItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
