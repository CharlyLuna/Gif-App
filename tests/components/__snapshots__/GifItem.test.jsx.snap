// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem tests should match the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Some title"
      src="https://some-random-url.com/img"
    />
    <p>
      Some title
    </p>
  </div>
</div>
`;
